kafka-admin/src/main/java/io/strimzi/admin/kafka/admin/handlers/TopicCreateHandler.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.admin.kafka.admin.handlers;
+
+import io.strimzi.admin.kafka.admin.AdminClientProvider;
+import io.strimzi.admin.kafka.admin.model.Types;
+import io.vertx.core.Promise;
+import io.vertx.ext.web.handler.graphql.VertxDataFetcher;
+import io.vertx.kafka.admin.NewTopic;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TopicCreateHandler {
+
+    public static VertxDataFetcher createTopic(AdminClientProvider acp) {
+        VertxDataFetcher<Types.CreateTopicInput> dataFetcher = new VertxDataFetcher<>((environment, prom) -> {
+
+            NewTopic newTopic = new NewTopic();
+            Map<String, Object> input = environment.getArgument("input");
+            newTopic.setName(input.get("name").toString());
+            Map<String, String> config = new HashMap<>();
+            Map<String, Object> configObject = (Map<String, Object>) input.get("config");
+            Types.CreateTopicInput createTopicInput = new Types.CreateTopicInput();
+            createTopicInput.setName(input.get("name").toString());
+            Types.CreateOrMutateTopicConfigInput createOrMutateTopicConfigInput = new Types.CreateOrMutateTopicConfigInput();
+
+            if (configObject.get("partitionCount") != null) {
+                String val = configObject.get("partitionCount").toString();
+                createOrMutateTopicConfigInput.setPartitionCount(Long.parseLong(val));
+                newTopic.setNumPartitions(Integer.parseInt(val));
+            }
+            if (configObject.get("replicationFactor") != null) {
+                String val = configObject.get("replicationFactor").toString();
+                createOrMutateTopicConfigInput.setReplicationFactor(Long.parseLong(val));
+                newTopic.setReplicationFactor(Short.parseShort(val));
+            }
+            if (configObject.get("pairs") != null) {
```

Comment:
is this the topic configuration options, why not calling "config" instead of "pairs"?

Reply:
The query would look like this (nested `config`). I am not sure we like that.
```
query {
  topicDescription(name: "topicX") {
    name
    config {
      partitionCount
      replicationFactor
      config {
        key
        value
      }
    }
  }
}
```